pkg/runtime/convert.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Open Component Model contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package runtime
+
+type Converter[T TypedObject] interface {
+	ConvertFrom(object T) (TypedObject, error)
+	ConvertTo(object interface{}) (T, error)
+}
+
+// ConvertWrapper wraps a typed converter into a generic
+// converter to fulfill the more generic interface.
+type ConvertWrapper[T TypedObject] struct {
+	converter Converter[T]
+}
+
+func (c *ConvertWrapper[T]) ConvertTo(object interface{}) (TypedObject, error) {
+	return c.converter.ConvertTo(object)
+}
+
+func NewConvertWrapper[T TypedObject](c Converter[T]) TypedObjectConverter {
+	return &ConvertWrapper[T]{converter: c}
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
```

Comment:
FormatVersion, ConvertedType  and VersionedType should be unified (in a follow-up PR). 
If the external format version is equal to the internal format version, the conversion should be the identity. 

Such a change should reduce the number and nesting of interfaces.